### PR TITLE
Feature/bai 2563 new endpoint entries unapproved releases

### PR DESCRIPTION
**Pull request by ON93754**

### CODE REVIEW

File: backend/src/connectors/metrics/base.ts [hunk 1]
```diff
@@ -454,6 +458,218 @@ async function calculateModelsMissingReleases(org?: string): Promise<NoReleasesC
...
+  let totalUnapprovedReleases = 0
+
+  for (const model of models) {
+    const unapprovedReleases = unapprovedByModel.get(model.id)
+
+    if (unapprovedReleases === undefined || unapprovedReleases.size === 0) {
+      continue
+    }
+
+    const sortedReleases = [...unapprovedReleases].sort()
```
Comment (PE39806) [inline backend/src/connectors/metrics/base.ts]: This will not sort semantically (e.g. `['1.0.0', '2.0.0', '10.0.0'].sort()` would give `['1.0.0', '10.0.0', '2.0.0']`)
```suggestion
    const sortedReleases = [...unapprovedReleases].sort((a, b) =>
      semver.compare(a, b),
    )
```
plus `import semver from 'semver'`

File: backend/src/connectors/metrics/base.ts [hunk 2]
```diff
@@ -454,6 +458,218 @@ async function calculateModelsMissingReleases(org?: string): Promise<NoReleasesC
...
+      })
+    }
+
+    const entry = releaseRoleMap.get(key)!
+
+    entry.requiredRoles.add(review.role)
+
+    if (approvedReviewIds.has(review._id.toString())) {
+      entry.approvedRoles.add(review.role)
+    }
```
Comment (PE39806) [inline backend/src/connectors/metrics/base.ts:593]: I think this might get funky if different individuals with the same role give different reviews e.g. for some Release `1.0.0` it could have `Review User1 (MSRO) approved` but also `Review UserB (MSRO) pending`. I'm not quite sure how we resolve this normally but something to check and maybe add a comment here if it's expected behaviour.

File: backend/src/routes/v3/metrics/getUnapprovedComplianceMetrics.ts
Comment (PE39806) [inline backend/src/routes/v3/metrics/getUnapprovedComplianceMetrics.ts:1]: Add new `backend/test/routes/v3/metrics/getUnapprovedComplianceMetrics.spec.ts` file.

**Review by PE39806 — CHANGES_REQUESTED** [review]

File: backend/test/utils/version.spec.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { describe, expect, test } from 'vitest'
+
+import { sortSemvers } from '../../src/utils/version.js'
+
+describe('utils > version > sortSemvers', () => {
```
Comment (PE39806) [inline backend/test/utils/version.spec.ts:5]: I'd like another test to verify that this safely handles duplicate values e.g. `sortSemvers(['1.0.0', '1.0.0'])`.

**Review by PE39806 — CHANGES_REQUESTED** [review]

**Review by PE39806 — APPROVED** [review]

**Review by ARADDCC002 — APPROVED** [review]
